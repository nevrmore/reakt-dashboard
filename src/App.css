.round-square {
  margin: 5%;
}

.components {
  background-color: #252B32;
  border-radius   : 0 20px 20px 0px;
  min-height      : 200px;
  padding         : 30px;
}

@media only screen and (min-width: 768px) {

  /* tablets and desktop */
  .todo-main {
    padding-left: 0;
  }

  .football-main {
    padding-right: 0;
  }
}

@media only screen and (max-width: 767px) {
  .components {
    border-radius: 20px;
    padding      : 25px;
    margin-left  : 0px;
  }

  .todo-main {
    padding      : 0;
    margin-bottom: 20px;
  }

  .football-main {
    padding: 0;
  }
}

.scrollbar {
  margin-bottom: 5px;
}

#style-1::-webkit-scrollbar {
  width           : 5px;
  background-color: #1A1F24;

}

#style-1::-webkit-scrollbar-thumb {
  border-radius     : 10px;
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, .3);
  box-shadow        : inset 0 0 6px rgba(0, 0, 0, .3);
  background-color  : #121212;
}