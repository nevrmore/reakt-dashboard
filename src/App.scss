@import "./assets/scss/index.scss";

.app {
  .round-square {
    margin: 2% 5%;
  }

  .card {
    background-color: $black-perl;
  }

  .components {
    background-color: $cinder;
    border-radius   : 20px;
    min-height      : 200px;
    padding         : 25px;
    margin-left     : 0px;
    max-height      : 700px;

    @include md {
      border-radius: 0 20px 20px 0;
      min-height   : 200px;
      padding      : 30px;
      margin-left  : -15px;
    }
  }

  .right-side {
    .todo-main {
      margin-bottom: 20px;
      padding      : 0;

      @include lg {
        padding-right: 30px;
      }
    }

    .football-main {
      padding: 0px;

      @include lg {
        padding-right: 0;
      }
    }
  }
}