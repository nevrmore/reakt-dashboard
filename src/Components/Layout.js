import React from 'react';
import PropTypes from 'prop-types';

const WeatherWidget = React.lazy(() => import('./WeatherWidget/WeatherWidget'));
const TodoList = React.lazy(() => import('./TodoList/TodoList'));
const Football = React.lazy(() => import('./Football/Football'));

const Layout = (props) => {
  return (
    <div className="row">
      <div className="col-sm-4 p-0 left-side">
        <WeatherWidget />
      </div>
      <div className="row col-sm-8 right-side components">
        <div className="col-sm todo-main">
          <TodoList />
        </div>
        <div className="col-sm football-main">
          <Football />
        </div>
      </div>
    </div>
  );
};

Layout.propTypes = {

};

export default Layout;
