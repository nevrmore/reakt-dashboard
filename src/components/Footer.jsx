const Footer = () => {
	return (
		<footer className="text-muted">
			<div className="container">
				<p className="float-right">
					<a
						href="https://www.linkedin.com/in/wsameer/"
						rel="noopener noreferrer"
						target="_blank">
						Follow me on Linkedin
					</a>
				</p>
				<p>
					This React dashboard project is a Â©{' '}
					<a
						href="https://github.com/wsameer"
						rel="noopener noreferrer"
						target="_blank">
						Sameer Waskar
					</a>
					, but please fork a branch and customize it for yourself!
				</p>
			</div>
		</footer>
	);
};

export default Footer;
