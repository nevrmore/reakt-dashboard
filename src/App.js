import React, { Suspense } from 'react';
import './App.scss';
import BusyIndicator from './Components/BusyIndicator';
import Header from './Components/Header';
import Footer from './Components/Footer';
import Layout from './Components/Layout';

function App() {
  return (
    <Suspense fallback={<BusyIndicator />}>
      <div className="app container-fluid">
        <Header />
        <div className="round-square">
          <Layout />
          <Footer />
        </div>
      </div>
    </Suspense>
  );
};

export default App;
