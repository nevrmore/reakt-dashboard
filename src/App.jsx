import { Suspense } from 'react';
import './App.scss';
import BusyIndicator from './components/BusyIndicator';
import Football from './components/Football/Football';
import Footer from './components/Footer';
import Todo from './components/Todo/Todo';
import Weather from './components/Weather/Weather';
import PageLayout from './components/layout/PageLayout';

const App = () => {
	return (
		<Suspense fallback={<BusyIndicator />}>
			<PageLayout>
				<div className="row">
					<div className="col-sm-4 p-0 left-side">
						<Weather />
					</div>
					<div className="row col-8 right-side components">
						<Todo />
						<Football />
					</div>
				</div>
			</PageLayout>
			<Footer />
		</Suspense>
	);
};

export default App;
