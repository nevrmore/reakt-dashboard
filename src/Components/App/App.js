import React, { Suspense } from 'react';
import './App.scss';
import BusyIndicator from '../BusyIndicator';
import Layout from '../Layout/Layout';
import Footer from '../Footer';

function App() {
  return (
    <Suspense fallback={<BusyIndicator />}>
      <div className="app container-fluid p-0">
        <div className="round-square">
          <Layout />
          <Footer />
        </div>
      </div>
    </Suspense>
  );
};

export default App;
