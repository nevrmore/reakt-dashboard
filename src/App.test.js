import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';
import Layout from './Components/Layout';
import Footer from './Components/Footer';

describe('<App />', () => {
  let component;

  beforeEach(() => (component = shallow(<App />)));

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should contain Layout component', () => {
    expect(component.find(Layout)).toHaveLength(1);
  });

  it('should contain Footer component', () => {
    expect(component.find(Footer)).toHaveLength(1);
  });
});