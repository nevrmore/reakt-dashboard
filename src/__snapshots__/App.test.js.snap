// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<Suspense
  fallback={<BusyIndicator />}
>
  <div
    className="app container-fluid p-0"
  >
    <div
      className="round-square"
    >
      <Layout />
      <Footer />
    </div>
  </div>
</Suspense>
`;
